modules/core/include/opencv2/core/hal/intrin_vsx.hpp

```diff
@@ -38,6 +38,8 @@ struct v_uint8x16
     {}
     uchar get0() const
     { return vec_extract(val, 0); }
+    uchar get(const int& i) const
```

Comment:
Should be templated:

```
template<int i>
uchar get() const;
{
    return v_rotate_right<i>(*this).get0();
}
```

with this implementation for all other non-VSX SIMD backends.

---

Also we can add a separate function (`vec_extract<i>(a)` ?) instead of member function.

Reply:
ok so what about removing the member functions and implementing something like this on VSX?

```
#define OPENCV_HAL_IMPL_VSX_EXTRACT(_Tpvec, _Tp) \
template<int i> inline _Tp v_extract(_Tpvec v) { return vec_extract(v.val, i); }

OPENCV_HAL_IMPL_VSX_EXTRACT(v_uint8x16, uchar)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_int8x16, schar)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_uint16x8, ushort)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_int16x8, short)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_uint32x4, uint)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_int32x4, int)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_uint64x2, uint64)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_int64x2, int64)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_float32x4, float)
OPENCV_HAL_IMPL_VSX_EXTRACT(v_float64x2, double)```

I think it might be better to name it v_extract instead of vec_extract first because vec_extract is the name of the intrinsic and secondly because other HAL functions usually start with v_ or vx_ instead of vec_. 

For the other archs we can use the rotate approach. What do you think?